serde_codegen/src/attr.rs

```diff
@@ -212,13 +212,24 @@ pub struct FieldAttrs {
     skip_serializing_field: bool,
     skip_deserializing_field: bool,
     skip_serializing_if: Option<ast::Path>,
-    default_expr_if_missing: Option<P<ast::Expr>>,
+    default: FieldDefault,
     serialize_with: Option<ast::Path>,
     deserialize_with: Option<ast::Path>,
     ser_bound: Option<Vec<ast::WherePredicate>>,
     de_bound: Option<Vec<ast::WherePredicate>>,
 }
 
+/// Represents the default to use for a field when deserializing.
+#[derive(Debug, PartialEq)]
+pub enum FieldDefault {
+    /// Field must always be specified because it does not have a default.
+    None,
+    /// The default is given by `std::default::Default::default()`.
+    Default,
+    /// The default is given by this function.
+    Path(ast::Path),
```

Comment:
It would be also great to support literals and constants as well as functions, so you can actually write
```rust
#[serde(default="0.1")]
pub field: f32,
```